src/librustc_mir/build/expr/into.rs

```diff
@@ -156,11 +156,16 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                     }
 
                     // execute the body, branching back to the test
-                    let unit_temp = this.unit_temp.clone();
-                    let body_block_end = unpack!(this.into(&unit_temp, body_block, body));
+                    // FIXME(#30636): this should not create or request any sort of destination at
+                    // all.
```

Comment:
I'm not sure I agree with this FIXME. In trans-land, we have some special cases for "ignoring" the result and it's kind of a pain. I guess though that we could have a special mode here *just for blocks*, maybe.